pkg/check/gc/reserve.go

```diff
@@ -27,68 +38,135 @@ func CheckReserve(c *bee.Cluster, o Options) error {
 	if err != nil {
 		return fmt.Errorf("create batch: %w", err)
 	}
-	fmt.Printf("created batch with depth %d and amount %d\n", depth, 1)
-	time.Sleep(time.Second * 5)
+	fmt.Printf("created batch with depth %d and amount %d\n", depth, lowValue)
+	time.Sleep(o.PostageWait)
 
 	state, err := node.Client().ReserveState(ctx)
 	if err != nil {
 		return fmt.Errorf("reservestate: %w", err)
 	}
 	fmt.Println("reservestate:", state)
 
-	lowValueChunks := chunkBatch(rnd, adrs.Overlay, int64(o.StoreSize), origState.Radius)
+	lowValueChunks := chunkBatch(rnd, overlay, int64(o.CacheSize), origState.Radius)
 	for _, c := range lowValueChunks {
 		_, err := node.Client().UploadChunk(ctx, c.Data(), api.UploadOptions{BatchID: batch})
 		if err != nil {
 			return fmt.Errorf("low value chunk: %w", err)
 		}
 	}
-	fmt.Printf("Uploaded %d chunks with batch depth %d\n", len(lowValueChunks), depth)
+	fmt.Printf("Uploaded %d chunks with batch depth %d, amount %d, at radius %d\n", len(lowValueChunks), depth, lowValue, origState.Radius)
+
+	// upload higher radius chunks that should not be garbage collected
+	higherRadius := origState.Radius + 1
+	higherRadiusChunkCount := int64(float64(o.CacheSize) * 0.1)
+	lowValueHigherRadiusChunks := chunkBatch(rnd, overlay, higherRadiusChunkCount, higherRadius)
+	for _, c := range lowValueHigherRadiusChunks {
+		if _, err := client.UploadChunk(ctx, c.Data(), api.UploadOptions{BatchID: batch}); err != nil {
+			return fmt.Errorf("low value chunk: %w", err)
+		}
+	}
+	fmt.Printf("Uploaded %d chunks with batch depth %d, amount %d, at radius %d\n", len(lowValueHigherRadiusChunks), depth, lowValue, higherRadius)
 
-	// now create batch with higher value
-	_, err = node.Client().CreatePostageBatch(ctx, 2, depth, "test-label")
+	// STEP 2: create high value batch thats covers the size of the reserve which should trigger the garbage collection of the low value batch
+	highValueBatch, err := client.CreatePostageBatch(ctx, highValue, depth, "test-label")
 	if err != nil {
 		return fmt.Errorf("create batch: %w", err)
 	}
-	fmt.Printf("created batch with depth %d and amount %d\n", depth, 2)
-	time.Sleep(time.Second * 5)
+	fmt.Printf("created batch with depth %d and amount %d\n", depth, highValue)
+	time.Sleep(o.PostageWait)
 
 	state, err = node.Client().ReserveState(ctx)
 	if err != nil {
 		return fmt.Errorf("reservestate: %w", err)
 	}
 	fmt.Println("reservestate:", state)
 
-	hasCounter := 0
+	hasCount := 0
 	for _, c := range lowValueChunks {
-		has, _ := node.Client().HasChunk(ctx, c.Address())
-		if has {
-			hasCounter++
+		if has, _ := client.HasChunk(ctx, c.Address()); has {
+			hasCount++
 		}
 	}
 
-	fmt.Printf("retrieved low value chunks: %d, gc'd count: %d\n", hasCounter, len(lowValueChunks)-hasCounter)
+	fmt.Printf("retrieved low value chunks: %d, gc'd count: %d\n", hasCount, len(lowValueChunks)-hasCount)
 
-	if len(lowValueChunks) == hasCounter {
+	if len(lowValueChunks) == hasCount {
```

Comment:
let's have a concrete check here that it _has_ to equal to the amount of chunks we expect to be GCd